src/infer_response.cc

```diff
@@ -179,6 +179,47 @@ InferResponse::Error()
   return error_;
 }
 
+void
+InferResponse::SetNextResponseFuture(
+    std::promise<std::unique_ptr<InferResponse>>* promise)
+{
+  next_response_future_ =
+      std::make_unique<std::future<std::unique_ptr<InferResponse>>>(
+          promise->get_future());
+  ;
+}
+
+void
+InferResponse::ResetNextResponseFuture()
```

Comment:
Instead of manually setting these futures would it be possible to pass a `flags` argument in the constructor which automatically sets the proper futures? For example, if the flags says it is the final request the future would be null, otherwise it would be an empty promise?

Reply:
Yes. Now we are setting the future in the constructor.

Reply:
Can we delete this function now?

Reply:
Yes, deleted.